chain/client/src/view_client.rs

```diff
@@ -246,6 +250,62 @@ impl ViewClientActor {
         }
     }
 
+    /// Returns maintenance windows by account.
+    fn get_maintenance_windows(
+        &self,
+        account_id: AccountId,
+        genesis_config: GenesisConfig,
+    ) -> Result<MaintenanceWindowsView, near_chain::Error> {
+        let epoch_manager = EpochManager::new_from_genesis_config(
```

Comment:
you shouldn't be creating your own epoch manager, chainstore etc.

These are alreay available in this object (self.chain, self.runtime_adapter).

// When you create a ChainStore - it looks like you're trying to read the data directly from storage again.

Reply:
Thanks for pointing out this.  Could you help me to get the `EpochInfo` from `RuntimeAdaptor`? There is no `get_epoch_info` in the `EpochManagerAdaptor` trait.  I lack the background on this.  Thanks in advance.